src/main/java/com/minecolonies/coremod/colony/CitizenData.java

```diff
@@ -449,6 +450,29 @@ public void initEntityValues()
         citizen.getEntityData().set(DATA_JOB, getJob() == null ? "" : getJob().getJobRegistryEntry().getKey().toString());
         citizen.getEntityData().set(DATA_STYLE, colony.getTextureStyleId());
 
+        // Safety check that ensure the citizen its job matches the work building job
+        if (getJob() != null && workBuilding != null)
+        {
+            try
+            {
+                workBuilding.getModuleMatching(WorkerBuildingModule.class, m -> m.getJobEntry().equals(getJob().getJobRegistryEntry()));
```

Comment:
Btw, is there no "getOptionalModuleMatching" method atm?

Reply:
No, there's only overloads of these methods that do not have a predicate match in them.
I can theoretically use `.getModules()` and iterate all of them to find the first problematic module, but that would seem less than ideal.

Reply:
I changed up the logic a bit, is it better now?